packages/react/src/components/Forms/Select/index.tsx

```diff
@@ -0,0 +1,342 @@
+
+  // When component is controlled, this useEffect will update the local selected item.
+  useEffect(() => {
+    if (downshiftProps.initialSelectedItem !== undefined) {
+      return;
+    }
+    const optionToSelect = options.find(
+      (option) => itemToValue(option) === value
+    );
+    downshiftReturn.selectItem(optionToSelect ?? null);
+  }, [value]);
+
+  return (
+    <Field
+      state={state}
+      text={text}
+      rightText={rightText}
+      fullWidth={fullWidth}
+    >
+      <div
+        className={classNames("c__select", "c__select--" + state, {
+          "c__select--disabled": disabled,
+        })}
+      >
+        {/* We disabled linting for this specific line because we consider that the onClick props is only used for */}
+        {/* mouse users, so this do not engender any issue for accessibility. */}
+        {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
+        <div
+          className={classNames("c__select__wrapper", {
+            "c__select__wrapper--focus": downshiftReturn.isOpen && !disabled,
+          })}
+          {...downshiftReturn.wrapperProps}
+        >
+          {downshiftReturn.selectedItem && (
+            <input
+              type="hidden"
+              name={name}
+              value={itemToValue(downshiftReturn.selectedItem)}
+            />
+          )}
+          <LabelledBox
+            label={label}
+            labelAsPlaceholder={labelAsPlaceholder}
+            htmlFor={labelProps.htmlFor}
+            labelId={labelProps.id}
+          >
+            <div className="c__select__inner">
+              <div className="c__select__inner__value">{children}</div>
+              <div className="c__select__inner__actions">
+                {!disabled && downshiftReturn.selectedItem && (
+                  <>
+                    <Button
+                      color="tertiary"
+                      size="small"
+                      aria-label="clear selection"
+                      className="c__select__inner__actions__clear"
+                      onClick={() => {
+                        downshiftReturn.selectItem(null);
+                      }}
+                      icon={<span className="material-icons">close</span>}
+                    />
+                    <div className="c__select__inner__actions__separator" />
+                  </>
+                )}
+
+                <Button
+                  color="tertiary"
+                  size="small"
+                  className="c__select__inner__actions__open"
+                  icon={
+                    <span
+                      className={classNames("material-icons", {
+                        opened: downshiftReturn.isOpen,
+                      })}
+                    >
+                      arrow_drop_down
+                    </span>
+                  }
+                  disabled={disabled}
+                  {...downshiftReturn.toggleButtonProps}
+                />
```

Comment:
In reality it depends. If this field is a select ( in terms of Downshift ) those are empty because those props are passed to the wrapper ( the main box with the grey border ), in this case the button is only here for visual purpose, see the doc here where the arrow is just a plain span: https://www.downshift-js.com/use-select

Where if the field is a combo box ( searchable select in our terms ) this button indeed gets aria stuff, except `aria-label` that I missed, see the fixup 👌